app/models/division.rb

```diff
@@ -124,9 +125,14 @@ def parent_division_and_name
   end
 
   def generate_short_name
-    return if short_name.present? && Division.pluck(:short_name).exclude?(self.short_name)
+    # no change to short_name that is already saved and therefore uniq
+    return if self.short_name.present? && self.short_name == self.attribute_in_database(:short_name)
+
+    # accept a short_name that is being provided and is uniq
+    return if self.short_name.present? && Division.pluck(:short_name).exclude?(self.short_name)
```

Comment:
`self.` not needed here.

Reply:
Separately, a more efficient way to check for uniqueness is `Division.where(short_name: short_name).empty?`, which Rails will turn into an EXISTS query and thus you won't need to load all the existing short names into memory and check them with ruby `exclude?`